cmd/kubernetes.go

```diff
@@ -0,0 +1,818 @@
+		}
+		kubeLogger.SetHandler(log15.LvlFilterHandler(logLevel, l15h.CallerInfoHandler(log15.StderrHandler)))
+
+		// Read in post creation script
+		var postCreation []byte
+		var extraArgs []string
+		if podPostCreationScript != "" {
+			var err error
+			postCreation, err = ioutil.ReadFile(podPostCreationScript)
+			if err != nil {
+				die("--script %s could not be read: %s", podPostCreationScript, err)
+			}
+			// daemon runs from /, so we need to convert relative to absolute
+			// path *** and then pretty hackily, re-specify the option by
+			// repeating it on the end of os.Args, where the daemonization code
+			// will pick it up
+			pcsAbs, err := filepath.Abs(podPostCreationScript)
+			if err != nil {
+				die("--script %s could not be converted to an absolute path: %s", podPostCreationScript, err)
+			}
+			if pcsAbs != postCreationScript {
+				extraArgs = append(extraArgs, "--script")
+				extraArgs = append(extraArgs, pcsAbs)
+			}
+		} else {
+			podPostCreationScript = "nil.sh"
+		}
+
+		// first we need our working directory to exist
+		createWorkingDir()
+
+		// check to see if the manager is already running (regardless of the
+		// state of the pid file); we can't proxy if a manager is already up
+		jq := connect(1*time.Second, true)
+		if jq != nil {
+			die("wr manager on port %s is already running (pid %d); please stop it before trying again.", config.ManagerPort, jq.ServerInfo.PID)
+		}
+
+		// now check if there's a daemon running.
+		// If it is the forwarding must've died. (Closed laptop?)
+		// If so kill the now useless daemon. This avoids the 'resource unavaliable'
+		// daemonising error.
+		fmPidFile := filepath.Join(config.ManagerDir, "kubernetes_resources.fw.pid")
+		fmPid, fmRunning := checkProcess(fmPidFile)
+
+		if fmRunning {
+			info("killing stale daemon with PID %s.", fmPid)
+			stale, err := os.FindProcess(fmPid)
+			if err != nil {
+				warn("Failed to find process", "err", err)
+			}
+			errr := stale.Kill()
+			if errr != nil {
+				warn("Killing process returned error", "err", errr)
+			}
+		}
+
+		// later we will copy our server cert and key to the manager pod;
+		// if we don't have any, generate them now
+		err := internal.CheckCerts(config.ManagerCertFile, config.ManagerKeyFile)
+		if err != nil {
+			err = internal.GenerateCerts(config.ManagerCAFile, config.ManagerCertFile, config.ManagerKeyFile, config.ManagerCertDomain)
+			if err != nil {
+				die("could not generate certs: %s", err)
+			}
+			info("created a new key and certificate for TLS")
+		}
+
+		// we will spawn wr on the remote server we will create, which means we
+		// need to know the path to ourselves in case we're not in the user's
+		// $PATH
+		exe, err := osext.Executable()
+		if err != nil {
+			die("could not get the path to wr: %s", err)
+		}
+
+		// we then  need to rewrite it to always use the 'wr-linux'
+		// binary, in case we are deploying from a mac.
+		exe = filepath.Dir(exe) + linuxBinaryName
```

Comment:
Is this specific to you having created your mac binary as `wr-mac` or something? I don't think anyone else would be in that situation?

Reply:
Yep, it is. The linuxBinaryName is what gets executed from the files copied to the cluster. Currently it's just set to `wr`. The mac binary can be named whatever, e.g asdf:

```
➜  wr git:(test) ✗ mv wr-mac asdf
➜  wr git:(test) ✗ ./asdf k8s teardown
INFO[08-20|12:16:10] opening resource file with path: /Users/tb15/.wr_development/kubernetes_resources
EROR[08-20|12:16:10] could not open resource file with path: open /Users/tb15/.wr_development/kubernetes_resources: no such file or directory
➜  wr git:(test) ✗ ./asdf k8s deploy --debug
INFO[08-20|12:16:17] created a new key and certificate for TLS
INFO[08-20|12:16:17] authenticating using information from user's home directory clientlib=true
INFO[08-20|12:16:17] succesfully authenticated using information from user's home directory clientlib=true
INFO[08-20|12:16:17] please wait while wr is deployed to the cluster
INFO[08-20|12:16:30] wr manager remotely started in namespace, pod: goofy-johnson-wr wr-manager-b8c59f656-24q5x (10.233.120.121:11301)
INFO[08-20|12:16:30] wr's web interface can be reached locally at https://localhost:11302/?token=JLG6mapWIYwBSnLPEnDhbvJTVsrJwTWHsOH5gDuyBVY
```